src/App.tsx

```diff
@@ -15,36 +17,65 @@
   'Garlic',
 ];
 
+export enum SortType {
+  NONE = 'none',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
 export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
-        </button>
+  const [vissibleGoods, setVisibleGoods] = useState<string[]>(goodsFromServer);
+  const [sortGoods, setSortGoods] = useState<SortType>(SortType.NONE);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const sortedGoods = (field: SortType) => {
+    const updatedGoods = [...goodsFromServer];
+
+    if (field !== SortType.NONE) {
+      switch (field) {
+        case SortType.ALPHABET:
+          updatedGoods.sort((item1, item2) => item1.localeCompare(item2));
+          break;
+        case SortType.LENGTH:
+          updatedGoods.sort((item1, item2) => item1.length - item2.length);
+          break;
+        default:
+          break;
+      }
+    }
 
-        <button type="button" className="button is-success is-light">
-          Sort by length
-        </button>
+    if (isReversed) {
+      updatedGoods.reverse();
+    }
 
-        <button type="button" className="button is-warning is-light">
-          Reverse
-        </button>
+    setVisibleGoods(updatedGoods);
+    setSortGoods(field);
+  };
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
-      </div>
+  const reverseItems = () => {
+    const reversedGoods = [...vissibleGoods].reverse();
 
+    setVisibleGoods(reversedGoods);
+    setIsReversed(prevReverse => !prevReverse);
+  };
+
+  const reset = () => {
+    setVisibleGoods([...goodsFromServer]);
+    setSortGoods(SortType.NONE);
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <Buttons
+        sortBy={sortedGoods}
+        reset={reset}
+        sortField={sortGoods}
+        reverse={isReversed}
+        setReverse={reverseItems}
+      />
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        <GoodList goods={vissibleGoods} />
```

Comment:
The typo in `vissibleGoods` is repeated here. Ensure consistency by correcting it to `visibleGoods`.